### PR TITLE
Feature: Client/Server url schemas limitations for custom mobile apps

### LINKED ISSUE #5024: Request for Documentation: Document client/server url schemas limitations for custom mobile apps
Mattermost user `stu.doherty` from https://community-daily.mattermost.com has requested the following be documented:

```
Due to changes in our SAML auth flow any customer building their own version of the mobile client (either because they need to host the Push Proxy themselves, OR they have their own customizations they're applying to the client) will need to be aware of a few things:

1. If they want to allow users to simultaneously run the App Store versions of Mattermost, in addition to the custom company version, they will need to adapt the URL schemes used for the app in the build, as well as configure those schemes on the server using [App Custom URL Schemes](https://docs.mattermost.com/configure/configuration-settings.html#app-custom-url-schemes).  Method to adapt on mobile is still to be improved, but requires modifications in a few `mattermost-mobile` source files for now.
They also need to be aware that the bundleid for the application should not include `rnbeta` (_exact restriction to be confirmed_)
2. This same change would be required in a custom build of the Mattermost Desktop Application (_To be confirmed_).
3. The mobile and desktop custom clients would no longer be able to log into other Mattermost servers (unless they had the same custom app schema configuration change).

NOTE:  I believe this is not an issue for customers who have `inAppSessionAuth == true` in their MDM configuration [App Config Docs](https://docs.mattermost.com/deploy/mobile-appconfig.html)
```

See the original post [here](https://community-daily.mattermost.com/_redirect/pl/3hr7pgxttfgg8es18hxgmh3jgh).

_This issue was generated from [Mattermost](https://mattermost.com) using the [Doc Up](https://github.com/jwilander/mattermost-plugin-docup) plugin._

# QA Validation Criteria

In a generated docs preview:

- Visiting the page results in the URL Schema Limitations section displaying the intended text clarifications and updates.
- The link in this section takes users to the correct page

### PR DESCRIPTION
**Pull request by TomerPacific**
#### Summary
This pull request includes an update to the `Custom mobile app branding` section in the `source/deploy/build-custom-mobile-apps.rst` file. The update adds a new subsection detailing URL schema limitations for custom mobile app builds.

#### Ticket Link
Resolves #5024

### CODE REVIEW

File: source/deploy/build-custom-mobile-apps.rst [hunk 1]
```diff
@@ -54,6 +54,15 @@ Custom mobile app branding
 
 Ensure you select a unique app name that helps users distinguish your version from others, such as "<Your Company Name> Collaboration". See our `Brand and Visual Design Guidelines <https://handbook.mattermost.com/operations/operations/company-processes/publishing/publishing-guidelines/brand-and-visual-design-guidelines#name-usage-guidelines.html>`__ in our company Handbook for details.
 
+URL Schema Limitations
+~~~~~~~~~~~~~~~~~~~~~~
+If you are building your own version of Mattermost's mobile client, you need to be aware of the following things:
+
+- To allow users to simultaneously run the App Store versions of Mattermost, in addition to the custom company version, you will need to adapt the URL schemes used for the app in the build, as well as configure those schemes on the server using :doc:`App Custom URL Schemes </configure/site-configuration-settings>`
```
Comment (cwarnermm) [inline source/deploy/build-custom-mobile-apps.rst:61]:
```suggestion
- To allow users to simultaneously run the App Store versions of Mattermost, in addition to the custom company version, you will need to adapt the URL schemes used for the app in the build, as well as configure those schemes on the server using :ref:`App Custom URL Schemes <configure/site-configuration-settings:app custom url schemes>`
```

File: source/deploy/build-custom-mobile-apps.rst [hunk 2]
```diff
@@ -54,6 +54,15 @@ Custom mobile app branding
...
+- Be aware that the bundleid for the application should not include `rnbeta`
```
Comment (cwarnermm) [inline source/deploy/build-custom-mobile-apps.rst:62]:
```suggestion
- Be aware that the ``bundleid`` for the application should not include ``rnbeta``.
```

File: source/deploy/build-custom-mobile-apps.rst [hunk 3]
```diff
@@ -54,6 +54,15 @@ Custom mobile app branding
...
+- The same change would be required in a custom build of the Mattermost Desktop Application
```
Comment (cwarnermm) [inline source/deploy/build-custom-mobile-apps.rst:63]:
```suggestion
- The same change would be required in a custom build of the Mattermost desktop app.
```

File: source/deploy/build-custom-mobile-apps.rst [hunk 4]
```diff
@@ -54,6 +54,15 @@ Custom mobile app branding
...
+- The mobile and desktop custom clients would no longer be able to log into other Mattermost servers (unless they had the same custom app schema configuration change)
```
Comment (cwarnermm) [inline source/deploy/build-custom-mobile-apps.rst:64]:
```suggestion
- The mobile and desktop custom clients would no longer be able to log into other Mattermost servers (unless they had the same custom app schema configuration change applied).
```

**Review by enahum — APPROVED** [review]

**Review by larkox — APPROVED** [review]

**Comment by varghesejose2020** [comment]
@lindy65 please assign me

**Comment by varghesejose2020** [comment]
The changes in this PR have been tested in the preview environment and changes are working as expected. Adding screenshots for confirmation
Screenshot_2024_1023_224225
cc @lindy65

**Comment by lindy65** [comment]
Thanks for testing @varghesejose2020 - counted towards Hacktoberfest

cc @cwarnermm

**Pull request merged**